pkg/genericapiserver/destinations_test.go

```diff
@@ -0,0 +1,90 @@
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package genericapiserver
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage"
+	etcdstorage "k8s.io/kubernetes/pkg/storage/etcd"
+)
+
+func TestUpdateEtcdOverrides(t *testing.T) {
+	oldEtcdFn := newEtcdFn
+	defer func() { newEtcdFn = oldEtcdFn }()
+
+	testCases := []struct {
+		resource unversioned.GroupResource
+		servers  []string
+	}{
+		{
+			resource: unversioned.GroupResource{Group: api.GroupName, Resource: "resource"},
+			servers:  []string{"http://127.0.0.1:10000"},
+		},
+		{
+			resource: unversioned.GroupResource{Group: api.GroupName, Resource: "resource"},
+			servers:  []string{"http://127.0.0.1:10000", "http://127.0.0.1:20000"},
+		},
+		{
+			resource: unversioned.GroupResource{Group: extensions.GroupName, Resource: "resource"},
+			servers:  []string{"http://127.0.0.1:10000"},
+		},
+	}
+
+	defaultEtcdLocation := []string{"http://127.0.0.1"}
+	for _, test := range testCases {
+		actualEtcdConfig := etcdstorage.EtcdConfig{}
+		newEtcdFn = func(ns runtime.NegotiatedSerializer, storageVersion, memoryVersion unversioned.GroupVersion, etcdConfig etcdstorage.EtcdConfig) (etcdStorage storage.Interface, err error) {
+			actualEtcdConfig = etcdConfig
+			return nil, nil
+		}
+
+		defaultEtcdConfig := etcdstorage.EtcdConfig{
+			Prefix:     DefaultEtcdPathPrefix,
+			ServerList: defaultEtcdLocation,
+		}
+		storageDestinations := NewStorageDestinations(defaultEtcdConfig, api.Codecs, NewDefaultResourceEncodingConfig(), NewResourceConfig())
+		storageDestinations.SetEtcdLocation(test.resource, test.servers)
+
+		var err error
+		_, err = storageDestinations.Get(test.resource)
+		if err != nil {
+			t.Errorf("unexpected error %v", err)
+			continue
+		}
+		if !reflect.DeepEqual(actualEtcdConfig.ServerList, test.servers) {
+			t.Errorf("expected %v, got %v", test.servers, actualEtcdConfig.ServerList)
+			continue
+		}
+
+		_, err = storageDestinations.Get(unversioned.GroupResource{Group: api.GroupName, Resource: "unlikely"})
+		if err != nil {
+			t.Errorf("unexpected error %v", err)
+			continue
+		}
+		if !reflect.DeepEqual(actualEtcdConfig.ServerList, defaultEtcdLocation) {
+			t.Errorf("expected %v, got %v", defaultEtcdLocation, actualEtcdConfig.ServerList)
+			continue
+		}
+
+	}
+}
```

Comment:
My biggest concern is that such a crucial element as storage, has so low test coverage :(